### PR TITLE
[Calling][Feature] Leave call confirmation mode

### PR DESCRIPTION
**Pull request by DivyaMittal528**
## Purpose

This feature introduce an option into CallCompositeBuilder to enable/disable the "Leave Call?" confirmation dialog. If disable, the "End Call" button will directly end the call without prompt.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/Calling/CallingViewComponent/ControlBarViewModel.swift [hunk 1]
```diff
@@ -148,7 +148,11 @@ class ControlBarViewModel: ObservableObject {
     }
 
     func endCallButtonTapped() {
-        self.isConfirmLeaveListDisplayed = true
+        if self.leaveCallConfirmationMode == .alwaysEnabled {
+            self.isConfirmLeaveListDisplayed = true
```
Comment (iaulakh) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/Calling/CallingViewComponent/ControlBarViewModel.swift:152]: where is this value used?

Comment (DivyaMittal528) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/Calling/CallingViewComponent/ControlBarViewModel.swift:152]: isConfirmLeaveListDisplayed this value used to show leave call alert

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/CallingDemoView.swift [hunk 2]
```diff
@@ -250,6 +250,10 @@ extension CallingDemoView {
 
         var localizationConfig: LocalizationOptions?
         let layoutDirection: LayoutDirection = envConfigSubject.isRightToLeft ? .rightToLeft : .leftToRight
+        let barOptions = CallScreenControlBarOptions(leaveCallConfirmationMode:
```
Comment (pavelprystinka) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/CallingDemoView.swift:253]: We have two Views to launch call composite: this one, and CallingDemoViewController. Could you please add callScreenOptions there as well.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LeaveCallConfirmationMode.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import Foundation
+
+/// Enum defining options for leaving a call confirmation.
+public enum LeaveCallConfirmationMode {
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LeaveCallConfirmationMode.swift:9]: We can not use enum in the public API. Please see CallState or ParticipantRole for example.
Here is a documentation:
https://azure.github.io/azure-sdk/ios_design.html#enumerations

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

**Review by pavelprystinka — APPROVED** [review]

**Review by ShaunaSong — APPROVED** [review]

**Pull request merged**